### PR TITLE
Add types for mongration

### PR DESCRIPTION
**Pull request by soulthreads**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mongration/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    user?: string;
+    password?: string;
+    mongoUri?: string;
+    migrationCollection: string;
+    replicaSet?: string;
+}
+
+export interface MigrationResponse {
+    id: string;
+    status: string;
```
Comment (sheetalkamat) [inline types/mongration/index.d.ts]: Could the type be `"pl" | "skipped" | "error" | "rollback" | "rollback-error" | "not-run"`

File: types/mongration/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export class Migration {
+    constructor(dbConfig: DbConfig);
+    add: (paths: string | string[]) => void;
+    addAllFromPath: (path: string) => void;
```
Comment (sheetalkamat) [inline types/mongration/index.d.ts]: Do not see this any where on readme. https://github.com/awapps/mongration#adding-steps-to-migration is this intended to be exposed?

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/mongration/index.d.ts (hunk 1 above)
Comment (soulthreads) [reply types/mongration/index.d.ts]: That's a good suggestion, thanks

File: types/mongration/index.d.ts (hunk 2 above)
Comment (soulthreads) [reply types/mongration/index.d.ts]: Yeah, they have it in their samples: https://github.com/awapps/mongration/blob/master/samples/03-folder-scan/migration.js#L14

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**